packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart

```diff
@@ -185,180 +185,157 @@ class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget>
+    super.didUpdateWidget(oldWidget);
+    _updateObjectGroupManager();
+    _updateProperties();
+  }
+
+  @override
+  void dispose() {
+    entranceController.dispose();
+    inspectorController.removeSelectionListener(_onInspectorSelectionChanged);
+    super.dispose();
+  }
+
+  /// required for getting all information required for visualizing Flex layout
+  Future<FlexLayoutProperties> fetchFlexLayoutProperties() async {
+    objectGroupManager?.cancelNext();
+    final nextObjectGroup = objectGroupManager.next;
+    final node = await nextObjectGroup.getDetailsSubtreeWithRenderObject(
+      getRoot(selectedNode),
+      subtreeDepth: 1,
     );
+    if (!nextObjectGroup.disposed) {
+      assert(objectGroupManager.next == nextObjectGroup);
+      objectGroupManager.promoteNext();
+    }
+    return FlexLayoutProperties.fromDiagnostics(node);
   }
 
-  Widget _visualizeChild({
-    LayoutProperties childProperties,
-    Color backgroundColor,
-    Color borderColor,
-    Color textColor,
-    Size renderSize,
-    Offset renderOffset,
-  }) {
-    Widget buildEntranceAnimation(BuildContext context, Widget child) {
-      final vertical = properties.isMainAxisVertical;
-      final horizontal = properties.isMainAxisHorizontal;
-      Size size = renderSize;
-      if (childProperties.flexFactor != null) {
-        size = SizeTween(
-          begin: Size(
-            horizontal ? minRenderWidth - entranceMargin : renderSize.width,
-            vertical ? minRenderHeight - entranceMargin : renderSize.height,
-          ),
-          end: renderSize,
-        ).evaluate(expandedEntrance);
-      }
-      return Opacity(
-        opacity: min([allEntrance.value * 5, 1.0]),
-        child: Padding(
-          padding: EdgeInsets.symmetric(
-            horizontal: (renderSize.width - size.width) / 2,
-            vertical: (renderSize.height - size.height) / 2,
-          ),
-          child: child,
-        ),
+  String id(RemoteDiagnosticsNode node) => node?.dartDiagnosticRef?.id;
+
+  void _onInspectorSelectionChanged() async {
+    if (!StoryOfYourFlexWidget.shouldDisplay(selectedNode)) {
+      return;
+    }
+    final prevRootId = id(_properties?.node);
+    final newRootId = id(getRoot(selectedNode));
+    final shouldFetch = prevRootId != newRootId;
+    FlexLayoutProperties newSelection = _properties;
+    if (shouldFetch) {
+      newSelection = await fetchFlexLayoutProperties();
+    }
+    setProperties(newSelection);
+  }
+
+  void updateHighlighted(FlexLayoutProperties newProperties) {
+    if (selectedNode.isFlex) {
+      highlighted = newProperties;
+    } else {
+      final idx = selectedNode.parent.childrenNow.indexOf(selectedNode);
+      if (idx != -1) highlighted = newProperties.children[idx];
+    }
+  }
+
+  void setProperties(FlexLayoutProperties newProperties) {
+    updateHighlighted(newProperties);
+    if (_properties == newProperties) {
+      // update highlighted child
+      setState(() {});
```

Comment:
The changes to the highlighted variable should be inside of setState calls.  Doing that will make this setState redundant.